modules/microshift-cni.adoc

```diff
@@ -0,0 +1,304 @@
+//https://docs.openshift.com/container-platform/4.11/networking/ovn_kubernetes_network_provider/about-ovn-kubernetes.html#nw-ovn-kubernetes-matrix_about-ovn-kubernetes
+
+//Q: are there immutable network settings we should tell users about?
+== {product-title} networking components and services overview
+This brief overview describes networking components and their operation in {product-title}. The `microshift-networking` RPM is a package that pulls in any networking related dependencies and certain systemd services to initialize networking, for example, the `microshift-ovs-init` systemd service.
+
+NetworkManager::
+NetworkManager is required by OVN-Kubernetes to set up the initial gateway bridge on the {product-title} node. NetworkManager and NetworkManager-OVS RPM packages are installed as dependencies to the `microshift-networking` RPM package. NetworkManager is configured to use the `keyfile` plugin by default and restarts after installation of the `microshift-networking` RPM package.
+//Q: Meaning the microshift networking RPM replaces the keyfile plugin?
+microshift-ovs-init::
+The `microshift-ovs-init.service` is installed by the `microshift-networking` RPM package as a one-shot systemd service. The 'microshift-ovs-init.service` executes the `configure-ovs.sh` script which uses NetworkManager commands to set up the OVS gateway bridge.
+
+OVN containers::
+OVN-Kubernetes cluster manifests are in the `microshift/assets/components/ovn` directory. Two OVN-Kubernetes daemon sets are rendered and applied by {product-title}.
+
+* *ovnkube-master*
+Includes the `northd`, `nbdb`, `sbdb` and `ovnkube-master` containers.
+
+* *ovnkube-node*
+The ovnkube-node includes the OVN-Controller container. After {product-title} boots, the OVN-Kubernetes daemon sets are deployed in the `openshift-ovn-kubernetes` namespace.
+
+Packaging::
+OVN-Kubernetes manifests and startup logic are built into {product-title}. The systemd services and configurations included in `microshift-networking` RPM are:
+
+* `/etc/NetworkManager/conf.d/microshift-nm.conf` for NetworkManager.service
+* `/etc/systemd/system/ovs-vswitchd.service.d/microshift-cpuaffinity.conf` for ovs-vswitchd.service
+* `/etc/systemd/system/ovsdb-server.service.d/microshift-cpuaffinity.conf`
+* `/usr/bin/configure-ovs-microshift.sh` for microshift-ovs-init.service
+* `/usr/bin/configure-ovs.sh` for microshift-ovs-init.service
+* `/etc/crio/crio.conf.d/microshift-ovn.conf` for CRI-O service
+
+== Configuring OVN-Kubernetes
+An OVN-Kubernetes config file must be written to /etc/microshift/ovn.yaml. {product-title} will use default OVN-Kubernetes configuration values if an OVN-Kubernetes config file is not customized.
+//Q: what are the default values?
+
+.Example `ovn.yaml` config file:
+
+[source, yaml]
+----
+ovsInit:
+  disableOVSInit: true
+  gatewayInterface: eth0
+  externalGatewayInterface: eth1
+mtu: 1300
+Configuring Host
+OVS bridge
+----
+//Q: is gatewayInterface enp1s0 or eth0?
```

Comment:
The interface name may vary on hardware or different (major) kernel versions, it also supports customization. The enp1s0 or eth0 is just an example, user has to fill the interface name according to their own environment.